### PR TITLE
Use Gradient Checkpointing Layer in Jamba & Blip Related Models

### PR DESCRIPTION
**Pull request by alex-jw-brooks**
Updates a few models to use the new `GradientCheckpointingLayer`. The models changed in this PR are leveraged by IBM model architectures (granite vision / bamba) or share code, either directly or through modular. Happy to submit a follow-up with other models too if that would be helpful!

Updated models:
- jamba
- bamba
- blip
- blip2
- instructblip
- instructblipvideo

@ArthurZucker

### CODE REVIEW

File: src/transformers/models/instructblip/modeling_instructblip.py [hunk 1]
```diff
@@ -885,31 +879,22 @@ def forward(
...
-                    attention_mask,
-                    layer_head_mask,
-                    encoder_hidden_states,
-                    encoder_attention_mask,
-                    past_key_value,
-                    output_attentions,
-                    query_length,
+            if getattr(self.config, "gradient_checkpointing", False) and self.training and use_cache:
+                logger.warning(
+                    "`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`..."
```
Comment (ArthurZucker) [inline src/transformers/models/instructblip/modeling_instructblip.py:884]: Nice, I think the gradient checkpoint layer does should get `use_cache` as input and fallback to not do GC ! 🤗

**Review by ArthurZucker — APPROVED** [review]
Thanks! Happy to merge, unless you have time to move the warning to the layer itself! might be a bit too big

File: src/transformers/models/instructblip/modeling_instructblip.py (hunk 1 above)
Comment (alex-jw-brooks) [reply src/transformers/models/instructblip/modeling_instructblip.py:884]: Awesome, thanks for merging this @ArthurZucker! I agree, I'll open a follow-up PR to take `use_cache` in the gcp layer and move the warning into it 😄
